src/sentry/static/sentry/app/components/events/interfaces/spans/spanBar.tsx

```diff
@@ -694,244 +728,145 @@ const CursorGuide = styled('div')`
 
   transform: translateX(-50%);
 
-  height: ${SPAN_ROW_HEIGHT}px;
+  height: 100%;
 `;
 
 export const DividerLine = styled('div')`
   position: absolute;
-  height: ${SPAN_ROW_HEIGHT}px;
-
+  height: 100%;
+  width: 1px;
   transform: translateX(-50%);
 
+  /* increase hit target */
+  border-width: 0 5px;
+  border-color: rgba(0, 0, 0, 0);
+  border-style: solid;
+  box-sizing: content-box;
+  background-clip: content-box;
+
   background-color: #cdc7d5;
   z-index: ${zIndex.dividerLine};
 
   &.hovering {
-    width: 4px !important;
+    width: 3px;
     cursor: col-resize;
   }
-
-  ${({hovering}: {hovering: boolean}) => {
-    if (!hovering) {
-      return 'width: 2px;';
-    }
-
-    return `
-      width: 4px;
-      cursor: col-resize;
-      `;
-  }};
 `;
 
 const SpanBarTitleContainer = styled('div')`
   display: flex;
   align-items: center;
-
-  height: ${SPAN_ROW_HEIGHT}px;
+  height: 100%;
   position: absolute;
   left: 0;
   top: 0;
   width: 100%;
+  user-select: none;
 `;
 
 const SpanBarTitle = styled('div')`
   position: relative;
-  top: 0;
-
-  height: ${SPAN_ROW_HEIGHT}px;
-  line-height: ${SPAN_ROW_HEIGHT}px;
-
-  color: #4a3e56;
-  font-size: 12px;
-
-  user-select: none;
-
+  height: 100%;
+  color: ${p => p.theme.gray4};
+  font-size: ${p => p.theme.fontSizeSmall};
   white-space: nowrap;
-
   display: flex;
+  flex: 1;
   align-items: center;
 `;
 
 const SpanTreeTogglerContainer = styled('div')`
   position: relative;
-  top: 0;
-
   height: 15px;
-
-  max-width: 40px;
   width: 40px;
-  min-width: 40px;
-
-  margin-right: 8px;
-
+  min-width: 40px; /* annoying flex thing */
+  margin-right: ${space(1)};
   z-index: ${zIndex.spanTreeToggler};
-
-  user-select: none;
-
   display: flex;
   justify-content: flex-end;
 `;
 
-const SpanTreeToggler = styled('div')`
-  position: relative;
+const getTogglerTheme = ({isExpanded}) => {
+  return isExpanded ? theme.button.default : theme.button.primary;
+};
 
-  white-space: nowrap;
+const getTogglerHoverTheme = ({isExpanded}) => {
+  return getTogglerTheme({isExpanded: !isExpanded});
+};
 
+const SpanTreeToggler = styled.div<{isExpanded: boolean}>`
```

Comment:
```suggestion
const SpanTreeToggler = styled('div')<{isExpanded: boolean}>`
```
I think we've been trying to use the function form of `styled()` so that the calls are the same for HTML elements and react components.

Reply:
yeah—I was a big advocate for that. But for some reason the functional version isn't playing very well if typescript. any suggestions?

Reply:
What kind of errors were you getting from typescript? I've been able to use the generic function but I had to include a wide selection props. https://github.com/getsentry/sentry/blob/b57861ac5318015c483bd24cab32db149426b056/src/sentry/static/sentry/app/components/well.tsx#L12-L14

Reply:
wow typescript  👀👀👀 

can you break this down for me? I understand what `omit` does, but what about the single amperstand? and the `keyof` thing I'm guessing transforms the `Props` types into something React can ingest?

Either way, I've found this pattern elsewhere and can just copy it. Thanks!!

Reply:
The `&` creates an [intersection type](https://www.typescriptlang.org/docs/handbook/advanced-types.html#intersection-types) which joins both two type definitions together.

The `keyof Props` expression creates a set of properties based on the keys in `Props`, as `Omit` requires a list of things to exclude.

Reply:
WOW thanks 😵  

Would it be helpful for me to make a follow up that attempts to push this into a helper somewhere? Seems like `props` is the only thing that needs to change regularly. 

I'm seeing that this pattern is already used in four places (soon to be five). Seems like a good time to intercept it?

Reply:
we could just have a bunch of these in a file somewhere, and do something like 

```js
import {withDivProps, withSvgProps, withImgProps} from 'app/utils/withProps';
```

Reply:
Having helpers for creating types would be awesome.

Reply:
We should make an eslint plugin that checks for the function form of `styled()`

Reply:
boom: https://astexplorer.net/#/gist/a7a9b7f9a40808ac7420924aff5949a8/ffd924bc95f9061e9d905dcff99f34593707fe7b